File: lab8/GCD.c
//GCD.c
//Aaron Hom
//awhom
//lab8
//Finds the GCD of 2 positive integers

#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <ctype.h>

int main(void){
  int a, b, a2, b2, c;
  int n = 1;
  int m = 2;
  char str[200];

  printf("Enter a positive number: ");

  while(n==1){
    scanf("%s", str); 
    if(isInteger(str)){
      sscanf(str, "%d", &a);        
      a2 = a;
      n = 3;
      break;
    }
    printf("Please enter a positive number: ");
    continue;
  }

printf("Enter another positive number: ");

  while(m==2){
    scanf("%s", str);
    if(isInteger(str)){
      sscanf(str, "%d", &b);     
      b2 = b;
      m = 4;
      break;
    }
    printf("Please enter a positive number: ");
    continue;
    }

  c = gcd(a2, b2);

  printf("The GCD of %d and %d is %d\n", a2, b2, c );
  
  return 0;
}

int gcd(int x, int y){
  int r;
    while(r > 0){
      r = x%y;
      x = y;
      y = r;
      if(r == 0){
	return x;
      }else{
      return 0;
    }
  }
}
int isInteger(char s[]){

  int i, h;
   
  if( s==NULL ) 
    return 0;
  h = strlen(s);

  if( h==0 ) 
    return 0;

  if( s[0]!='-' && s[0]!='+' && !isdigit(s[0]) )
    return 0;

  for(i=1; i<h; i++){
    if( !isdigit(s[i]) ) 
      return 0;
  }
  return 1;
}

File: prog3/GCD.java
//GCD.java
//Aaron Hom
//awhom
//Program3
//Takes user input of 2 positive integers and finds the greatest common denominator

import java.util.Scanner; //activate scanner
class GCD{ //initiate class 
   public static void main(String[] args){
       Scanner sc = new Scanner(System.in); //initiate scanner
       System.out.print("Enter a positive integer: "); //print text
       int a, b; //initiate variables
       while(true){ //starts  while loop (loops as long as statement is true)	
	   while(!sc.hasNextInt()){ //starts another while loop (loops as long as statement is not an int)
	       sc.next(); //deletes previously entered value
	       System.out.print("Please enter a positive integer: "); //prints text
           }
	   a = sc.nextInt(); //waits for user input
	   if(a < 0){ //tests if a is negative
	       System.out.print("Please enter a positive integer: "); //prints text
	   }else{ 
	       break; //breaks to outside statement bracket
	   }
       }
       System.out.print("Enter another positive integer: "); //prints text
       while(true){ //starts while loop (loops as long as statement is true)
	   while(!sc.hasNextInt()){ //starts another while loop (loops as long as statement is not true)
	       sc.next(); //deletes previously entered value
               System.out.print("Please enter a positive integer: "); //prints text
           }
	   b = sc.nextInt(); //waits for user input
	   if(b < 0){ //tests if b is negative
	       System.out.print("Please enter a positive integer: "); //prints text
 	   }else{  
	       break; //breaks to outside statement bracket
	   }
       }
       int r = 1; //initailizes variable 
       int a2 = a,b2 = b; //variables created to hold the orignial values of a and b
       while(r > 0){ //starts while loop (loops as long as r is greater than 0)
	   r = a % b; //r is "a modulus b"
	   a = b; //sets a equal to b value
	   b = r; //sets b equal to r value
	   if(r == 0){ //tests if r equals 0
	       System.out.println("The GCD of "+a2+" and "+b2+" is "+a); //prints the GCD of a and b      
         }
      }
   }
}
